projects/observability/src/pages/apis/service-detail/instrumentation/components/total-score.component.ts

```diff
@@ -0,0 +1,74 @@
+import { ChangeDetectionStrategy, Component, Input, OnChanges } from '@angular/core';
+import { ServiceInstrumentationService } from '../service-instrumentation.service';
+
+@Component({
+  styleUrls: ['./total-score.component.scss'],
+  selector: 'ht-service-instrumentation-total-score',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  template: `
+    <div class="service-instrumentation-total-score">
+      <ht-progress-circle
+        [percent]="serviceScore"
+        [colorLight]="scoreColors?.light"
+        [colorDark]="scoreColors?.dark"
+      ></ht-progress-circle>
+
+      <div>
+        <h4 class="heading">{{ this.getScoreLabel() }}</h4>
+
+        <p class="description">
+          {{ this.getDescriptionForScore() }}
+        </p>
+
+        <div class="legend">
+          <div class="dot">
+            <span [style.background-color]="'#dc3d43'"></span>
+            <label>0-49</label>
+          </div>
+
+          <div class="dot">
+            <span [style.background-color]="'#ffa01c'"></span>
+            <label>50-69</label>
+          </div>
+
+          <div class="dot">
+            <span [style.background-color]="'#3d9a50'"></span>
+            <label>70-100</label>
+          </div>
+        </div>
+      </div>
+    </div>
+  `
+})
+export class TotalScoreComponent implements OnChanges {
+  @Input()
+  public serviceScore: number = 0;
+
+  public scoreColors: { light: string; dark: string } | undefined;
+
+  public constructor(private readonly serviceInstrumentationService: ServiceInstrumentationService) {}
+
+  public ngOnChanges(): void {
+    this.scoreColors = this.serviceInstrumentationService.getColorForScore(this.serviceScore);
+  }
+
+  public getScoreLabel(): string {
+    return this.serviceInstrumentationService.getLabelForScore(this.serviceScore);
+  }
+
+  public getDescriptionForScore(): string {
```

Comment:
Umm, for uniformity purpose, can we move `getDescriptionForScore` method to the `serviceInstrumentationService`? Given we are doing `label` and `color` from there.

Reply:
Moved the `label` and `color` methods there since two components are using them: org-score and progress-circle. The descriptions will only be used in this component alone, not sure moving it to a service common to the entire page makes sense. Should I?

Reply:
I feel it will make the service accountable for all the information related to score. The disjointness between them here is a bit confusing. IMO, having it in service will make component cleaner and lean.

Reply:
✅ Sure. Moved the method to common service.